pkg/logging/logging.go

```diff
@@ -17,83 +17,47 @@
 package logging
 
 import (
-	"context"
 	"errors"
 	"fmt"
-	"os"
-	"path/filepath"
-	"strconv"
 
 	"github.com/containerd/containerd/runtime/v2/logging"
-	"github.com/containerd/nerdctl/pkg/logging/jsonfile"
-	units "github.com/docker/go-units"
-	"github.com/natefinch/lumberjack"
 )
 
 const (
 	// MagicArgv1 is the magic argv1 for the containerd runtime v2 logging plugin mode.
 	MagicArgv1 = "_NERDCTL_INTERNAL_LOGGING"
 	MaxSize    = "max-size"
 	MaxFile    = "max-file"
+	DriverType = "logging-driver"
 )
 
+type Interface interface {
+	init(argsMap map[string]string) error
+	getLoggerFunc(argsMap map[string]string) (logging.LoggerFunc, error)
+}
+
 // Main is the entrypoint for the containerd runtime v2 logging plugin mode.
 //
 // Should be called only if argv1 == MagicArgv1.
 func Main(argsMap map[string]string) error {
-	fn, err := getLoggerFunc(argsMap)
+	if _, ok := argsMap[DriverType]; !ok {
+		return errors.New("the driverType is empty")
+	}
+	var driver Interface
+	// TODO: Impl more drivers such as local, journald, fluentd
+	if argsMap[DriverType] == "json-file" {
```

Comment:
preparing to other drivers, it more humain friendly if we use a switch case here